modules/apps/app-builder/app-builder-test/src/testFunctional/tests/AppBuilder.testcase

```diff
@@ -1290,6 +1290,79 @@ definition {
 		AppBuilderAdmin.validateItemNameOnTable(itemName = "123");
 	}
 
+	@description = "Create more than one Table View and choose between the two when deploying an App"
+	@priority = "5"
+	test DeployAppChoosingBetweenDifferentTableViews {
+
+		AppBuilderAdmin.addCustomObject(
+			customObjectName = "Custom Object Title",
+			isCreatingFormView = "true",
+			isCreatingViaPlusIcon = "true");
+
+		AppBuilderAdmin.createFormViewAndTableView(
+			fieldType = "Text",
+			formViewName = "Text Form View",
+			tableViewName = "My Table View 1");
+
+		AppBuilderAdmin.switchTab(tabName = "Form Views");
+
+		AppBuilderAdmin.createFormViewAndTableView(
+			fieldType = "Upload",
+			formViewName = "Upload Form View ",
+			tableViewName = "My Table View 2");
+
+		AppBuilderAdmin.switchTab(tabName = "Apps");
+
+		AppBuilderAdmin.add();
+
+		AppBuilderAdmin.setName(title = "My First App");
+
+		App.selectFormView(formViewName = "Text Form View");
+
+		App.selectTableView(tableViewName = "My Table View 2");
+
+		Button.click(button = "Previous");
+
+		App.selectTableView(tableViewName = "My Table View 1");
+
+		App.deployAppToggleSwitcher(deployOption = "Widget");
+
+		Button.click(button = "Deploy");
+
+		Navigator.openURL();
+
+		Portlet.addPG(portletName = "My First App");
+
+		Button.click(button = "New Entry");
+
+		Button.click(button = "Save");
```

Comment:
Hey Tai, I would suggest you some changes here:
1. Replace Button.click(button = "New Entry") by AppBuilderAdmin.add(), as well as Button.click(button = "Save") by AppBuilderAdmin.save();

2. Add a real entry here instead of save as empty, so you can validate if the data will be still shown when changing the app to the TableView2 (containing the same field).